packages/react-components/src/components/resource-explorers/explorers/alarm-explorer/internal-alarm-explorer.tsx

```diff
@@ -0,0 +1,197 @@
+  createDefaultTableUserSettings,
+  DEFAULT_ALARM_RESOURCE_NAME,
+  DEFAULT_PLURAL_ALARM_RESOURCE_NAME,
+  latestValueTimeCellRenderer,
+  latestValueCellRenderer,
+} from '../../constants/defaults';
+import { AlarmResourceWithLatestValue } from '../../types/resources';
+import {
+  DEFAULT_ALARM_TABLE_DEFINITION,
+  createDefaultLatestValuesTableDefinition,
+} from '../../constants/table-resource-definitions';
+import { DEFAULT_ALARM_DROP_DOWN_DEFINITION } from '../../constants/drop-down-resource-definitions';
+import { useUserCustomization } from '../../helpers/use-user-customization';
+import { TableResourceDefinition } from '../../types/table';
+import { useAlarms } from '../../../../hooks/useAlarms';
+import { isAlarmResource, transformAlarmData } from './transformAlarmData';
+import { createNonNullableList } from '../../../../utils/createNonNullableList';
+
+export function InternalAlarmExplorer({
+  iotSiteWiseClient,
+  iotEventsClient,
+  parameters = [],
+  shouldPersistUserCustomization = DEFAULT_SHOULD_PERSIST_USER_CUSTOMIZATION,
+  defaultPageSize = DEFAULT_DEFAULT_PAGE_SIZE,
+  variant = DEFAULT_RESOURCE_EXPLORER_VARIANT,
+  resourceName = DEFAULT_ALARM_RESOURCE_NAME,
+  pluralResourceName = DEFAULT_PLURAL_ALARM_RESOURCE_NAME,
+  isAlarmDisabled = DEFAULT_IS_RESOURCE_DISABLED,
+  selectedAlarms = DEFAULT_SELECTED_RESOURCES,
+  onSelectAlarm = DEFAULT_ON_SELECT_RESOURCE,
+  selectionMode = DEFAULT_SELECTION_MODE,
+  tableResourceDefinition: customTableResourceDefinition,
+  defaultTableUserSettings: customDefaultTableUserSettings,
+  tableSettings: {
+    isTitleEnabled = DEFAULT_IS_TABLE_ENABLED,
+    isSearchEnabled = DEFAULT_IS_TABLE_SEARCH_ENABLED,
+    isFilterEnabled: isTableFilterEnabled = DEFAULT_IS_TABLE_FILTER_ENABLED,
+    isUserSettingsEnabled = DEFAULT_IS_TABLE_USER_SETTINGS_ENABLED,
+  } = {},
+  dropDownResourceDefinition = DEFAULT_ALARM_DROP_DOWN_DEFINITION,
+  dropDownSettings: { isFilterEnabled: isDropDownFilterEnabled = false } = {},
+  ariaLabels,
+  description = '',
+  timeZone,
+  significantDigits,
+}: AlarmExplorerProps) {
+  const requests = useMemo(
+    () =>
+      isAlarmByAssetRequestParameters(parameters)
+        ? parameters.map((p) => ({ assetId: p.assetId }))
+        : parameters.map((p) => ({ assetModelId: p.assetModelId })),
+    [parameters]
+  );
+
+  const alarms = useAlarms({
+    iotEventsClient,
+    iotSiteWiseClient,
+    requests,
+    transform: transformAlarmData,
+  });
+
+  const alarmResources = useMemo(
+    () => alarms.map(({ resource }) => resource).filter(isAlarmResource),
+    [alarms]
+  );
+
+  const isLoading = useMemo(
+    () => alarms.some(({ status }) => status.isLoading),
+    [alarms]
+  );
+
+  const error = useMemo(() => {
+    const errors = createNonNullableList(
+      alarms.flatMap(({ status }) => status.errors)
+    );
+    if (errors.length === 0) return null;
+
+    const message = errors.map((e) => e.message).join('; ');
```

Comment:
I talked with Tracy, and we decided that for now we will not actually return any content in the `errors` field. We may rename that back to `error` and decide how we want to handle aggregated errors as a team best practice. Instead we should just check `isError` and throw a generic error message like: `"Error loading SiteWise alarms for asset ${assetId}"`

We are logging all of the errors in useAlarms anyway.

Reply:
Cool. can modify this.